systemtest/src/main/java/io/strimzi/systemtest/logs/TestLogCollector.java

```diff
@@ -0,0 +1,267 @@
+        dateTimeFormatter = dateTimeFormatter.withZone(ZoneId.of("GMT"));
+        CURRENT_DATE = dateTimeFormatter.format(LocalDateTime.now());
+    }
+
+    /**
+     * TestLogCollector's constructor
+     */
+    public TestLogCollector() {
+        this.logCollector = defaultLogCollector();
+    }
+
+    /**
+     * Method for creating default configuration of the {@link LogCollector}.
+     * It provides default list of resources into the builder and configures the required {@link KubeClient} and
+     * {@link Kubectl}
+     *
+     * @return  {@link LogCollector} configured with default configuration for the tests
+     */
+    @SuppressWarnings("deprecation") // Kafka Mirror Maker is deprecated
+    private LogCollector defaultLogCollector() {
+        List<String> resources = new ArrayList<>(List.of(
+            TestConstants.SECRET.toLowerCase(Locale.ROOT),
+            TestConstants.DEPLOYMENT.toLowerCase(Locale.ROOT),
+            TestConstants.CONFIG_MAP.toLowerCase(Locale.ROOT),
+            Kafka.RESOURCE_SINGULAR,
+            KafkaNodePool.RESOURCE_SINGULAR,
+            KafkaConnect.RESOURCE_SINGULAR,
+            KafkaConnector.RESOURCE_SINGULAR,
+            KafkaBridge.RESOURCE_SINGULAR,
+            KafkaMirrorMaker.RESOURCE_SINGULAR,
+            KafkaMirrorMaker2.RESOURCE_SINGULAR,
+            KafkaRebalance.RESOURCE_SINGULAR,
+            KafkaTopic.RESOURCE_SINGULAR,
+            KafkaUser.RESOURCE_SINGULAR
+        ));
+
+        if (Environment.isOlmInstall()) {
+            resources.addAll(List.of(
+                TestConstants.OPERATOR_GROUP.toLowerCase(Locale.ROOT),
+                TestConstants.SUBSCRIPTION.toLowerCase(Locale.ROOT),
+                TestConstants.INSTALL_PLAN.toLowerCase(Locale.ROOT),
+                TestConstants.CSV.toLowerCase(Locale.ROOT)
+            ));
+        }
+
+        return new LogCollectorBuilder()
+            .withKubeClient(new KubeClient())
+            .withKubeCmdClient(new Kubectl())
+            .withRootFolderPath(Environment.TEST_LOG_DIR)
+            .withNamespacedResources(resources.toArray(new String[0]))
+            .build();
+    }
+
+    /**
+     * Method that checks existence of the folder on specified path.
+     * From there, if there are no sub-dirs created - for each of the test-case run/re-run - the method returns the
+     * full path containing the specified path and index (1).
+     * Otherwise, it lists all the directories, filtering all the folders that are indexes, takes the last one, and returns
+     * the full path containing specified path and index increased by one.
+     *
+     * @param rootPathToLogsForTestCase     complete path for test-class/test-class and test-case logs
+     *
+     * @return  full path to logs directory built from specified root path and index
+     */
+    private String checkPathAndReturnFullRootPathWithIndexFolder(String rootPathToLogsForTestCase) {
```

Comment:
instead of string and string join with "/" path delimiter, use `Paths` and methods resolve etc please

Reply:
Good point, thanks.